libsolidity/interface/CompilerStack.cpp

```diff
@@ -1548,6 +1549,9 @@ string CompilerStack::createMetadata(Contract const& _contract, bool _forIR) con
 	if (m_revertStrings != RevertStrings::Default)
 		meta["settings"]["debug"]["revertStrings"] = revertStringsToString(m_revertStrings);
 
+	meta["settings"]["metadata"]["append"] =
```

Comment:
Also keeping in line with `m_metadataLiteralSources`, these fields are not present unless set to non-default values. Not sure whether that is a good approach or not, but mentioning it.